src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -686,6 +686,24 @@ public function getInvoicesAction(int $orderId)
         ]);
     }
 
+    /**
+     * @AdminSecurity("is_granted(['create', 'update'], request.get('_legacy_controller'))", redirectRoute="admin_orders_index")
+     *
+     * @param int $orderId
+     */
+    public function getDocumentsAction(int $orderId)
+    {
+        /** @var OrderForViewing $orderForViewing */
+        $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
+
+        return $this->json([
+            'total' => count($orderForViewing->getDocuments()->getDocuments()),
+            'html' => $this->render('@PrestaShop/Admin/Sell/Order/Order/Blocks/View/documents.html.twig', [
```

Comment:
Is it not possible to ouput json rather than html ?

Reply:
You mean output `orderForViewing` as raw JSON data and handle the rendering on javascript side ?
This would be a very big change in the scope of the PR

Reply:
Affirmative. We need to do it before beta, because we can't change it after this release.

Reply:
I'm against it, just like I said for https://github.com/PrestaShop/PrestaShop/pull/18204/files#r402206127 😉but I see you saw it too
I started the discussion on slack 😄

Reply:
After discussion, it's :ok_hand: for me.